File: ruby/rna-transcription/rna_transcription.rb
# Given a DNA strand(String), returns its RNA complement(String)
class Complement
  DNA_TO_RNA = { 'G' => 'C', 'C' => 'G', 'T' => 'A', 'A' => 'U' }.freeze

  def self.of_dna(string)
    # check for invalid nucleotdes
    return '' if string[/[^#{DNA_TO_RNA.keys.join}]/]
    string.chars.collect do |nucleotide|
      DNA_TO_RNA[nucleotide]
    end.join
  end
end

module BookKeeping
  VERSION = 4
end

File: python/rna-transcription/rna_transcription.py
DNA_TO_RNA = { "G": "C", "C": "G", "T": "A", "A": "U" }

def to_rna(dna_string):
    rna_string = []
    for each in dna_string:
        if each not in DNA_TO_RNA:
            return ""
        else:
            rna_string.append(DNA_TO_RNA[each])
    return "".join(rna_string)
